truffle/src/main/java/org/jruby/truffle/runtime/ModuleOperations.java

```diff
@@ -243,6 +243,30 @@ public static Object lookupClassVariable(RubyModule module, String name) {
             }
         }
 
+        // If singleton class, check attached module as well.
+        if (module instanceof RubyClass) {
+            RubyClass klass = (RubyClass) module;
+
+            if (klass.isSingleton() && klass.getAttached() != null) {
+
+                // Look in the attached module.
+                value = klass.getAttached().getClassVariables().get(name);
+
+                if (value != null) {
+                    return value;
+                }
+
+                // Look in the attached module's ancestors.
+                for (RubyModule ancestor : klass.getAttached().parentAncestors()) {
+                    value = ancestor.getClassVariables().get(name);
+
+                    if (value != null) {
+                        return value;
+                    }
+                }
```

Comment:
We should not check both ancestry chains. MRI lookup is at https://github.com/ruby/ruby/blob/badb4de72a08b199f1a63864531d47beafca818b/variable.c#L2355-L2384.
So, I would add the `if attached` above the ancestors lookup, and in that `if` do the lookup on the attached module and set module to the attached module.